src/plugin.js

```diff
@@ -1152,29 +1207,31 @@ videojs.HlsHandler.prototype.drainBuffer = function() {
         code: 4
       });
     } else if (!segment.key.bytes) {
-
       // waiting for the key bytes, try again later
       return;
     } else if (segmentInfo.decrypter) {
-
       // decryption is in progress, try again later
       return;
-    } else {
-
-      // if the media sequence is greater than 2^32, the IV will be incorrect
-      // assuming 10s segments, that would be about 1300 years
-      segIv = segment.key.iv || new Uint32Array([0, 0, 0, mediaIndex + playlist.mediaSequence]);
-
-      // create a decrypter to incrementally decrypt the segment
-      decrypter = new videojs.Hls.Decrypter(segmentInfo.encryptedBytes,
-                                            segment.key.bytes,
-                                            segIv,
-                                            function(err, bytes) {
-                                              segmentInfo.bytes = bytes;
-                                            });
-      segmentInfo.decrypter = decrypter;
-      return;
     }
+    // if the media sequence is greater than 2^32, the IV will be incorrect
```

Comment:
You removed the else which is possible since all of the other cases explicitly return. We tend to keep the else (even if it is redundant) just to make the control-flow explicit. Also it protects us if, for some reason, one of the previous clauses are changed to no longer return.